sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -159,28 +161,64 @@ class FileStreamSource(
 
   /**
    * If the source has a metadata log indicating which files should be read, then we should use it.
-   * We figure out whether there exists some metadata log only when user gives a non-glob path.
+   * Only when user gives a non-glob path that will we figure out whether the source has some
+   * metadata log
+   *
+   * None        means we don't know at the moment
+   * Some(true)  means we know for sure the source DOES have metadata
+   * Some(false) means we know for sure the source DOSE NOT have metadata
```

Comment:
( some notes here since the changes are not trival )

here we're using this `sourceHasMetadata` to indicate whether we know for sure the source has metadata, as stated in the source file comments:
- `None`        means we don't know at the moment
- `Some(true)`  means we know for sure the source DOES have metadata
- `Some(false)` means we know for sure the source DOSE NOT have metadata